src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs

```diff
@@ -2215,6 +2215,94 @@ public void AfterTargetsComeFromImport()
             }
         }
 
+        /// <summary>
+        /// Verifies that Directory.Solution.props and Directory.Solution.targets are imported by the generated project, that the import
+        /// can be disabled, and that you can specify a custom name for the projects.
+        /// </summary>
+        /// <param name="projectName">The name of the project to create.</param>
+        /// <param name="enable"><code>true</code> to have the functionality enabled, otherwise <code>false</code>.</param>
+        [Theory]
+        [InlineData("Directory.Solution.props", true)]
+        [InlineData("Directory.Solution.props", false)]
+        [InlineData("Directory.Solution.targets", true)]
+        [InlineData("Directory.Solution.targets", false)]
+        [InlineData("Custom.Directory.Solution.props", true)]
+        [InlineData("Custom.Directory.Solution.targets", true)]
+        public void DirectorySolutionPropsTest(string projectName, bool enable)
+        {
+            const string expectedPropertyValue = "ValueA";
+
+            string baseDirectory = Guid.NewGuid().ToString("N");
+
+            string solutionFilePath = ObjectModelHelpers.CreateFileInTempProjectDirectory(Path.Combine(baseDirectory, $"{Guid.NewGuid():N}.sln"), @"
```

Comment:
nit: TestEnvironment is preferred because it cleans up after itself and makes the tests more uniform

Reply:
No other test in this file is using `TestEnvironment`, do you think I should have this test stray from the rest?

Reply:
Personal decision I guess :) I prefer to break the look and feel of the surrounding code if I feel I'm replacing it with something better.

Reply:
I'm going to keep the file consistent